packages/lesswrong/server/rsvpToEvent.ts

```diff
@@ -60,3 +60,37 @@ addGraphQLResolvers({
     }
   }
 });
+
+addGraphQLMutation('CancelRSVPToEvent(postId: String, name: String, userId: String): Post');
+addGraphQLResolvers({
+  Mutation: {
+    async CancelRSVPToEvent(root: void, {postId, name, userId}: {postId: string, name: string, userId: string}, context: ResolverContext) {
+      const { currentUser } = context;
+      const post = await context.loaders.Posts.load(postId);
+
+      if (currentUser?._id !== userId && currentUser?._id !== post.userId) {
+        throw new Error("user does not have permission to remove rsvps of this userId")
+      }
+
+      // eslint-disable-next-line no-console
+      console.log("old rsvps", post.rsvps)
+      let rsvps = post.rsvps.filter(rsvp => rsvp.name !== name) || []
```

Comment:
nit: should be `const`, since I don't see this getting reassigned anywhere?  Also, not sure the `||` is doing anything, since `filter` should never return a false-y value - an empty array at worst